SharpPcap/CaptureEventArgs.cs

```diff
@@ -26,18 +26,29 @@ namespace SharpPcap
     /// <summary>
     /// Capture event arguments
     /// </summary>
-    public class CaptureEventArgs : EventArgs
+    public ref struct CaptureEventArgs
     {
+        private RawCapture rawCapture;
+
         /// <summary>
         /// Packet that was captured
         /// </summary>
-        public RawCapture Packet { get; }
+        public RawCapture Packet {
+            get
+            {
+                return rawCapture ?? (rawCapture = new RawCapture(Device, Header, Data));
+            }
+        }
 
         /// <summary>
         /// Device this EventArgs was generated for
         /// </summary>
         public ICaptureDevice Device { get; }
 
+        public ICaptureHeader Header { get; }
+
+        public Span<byte> Data { get; }
```

Comment:
We don't want users to alter the Data content
```suggestion
        public ReadOnlySpan<byte> Data { get; }
```